build.gradle

```diff
@@ -32,19 +37,23 @@ subprojects {
   }
 
   // Common dependencies
-  ext.springVersion = '4.3.12.RELEASE'
-  ext.springBootVersion = '1.5.9.RELEASE'
+  ext.compileWithSpringVersion = '5.0.6.RELEASE'
+  ext.compileWithBootVersion = '2.0.2.RELEASE'
+
+  ext.testWithSpringVersion = '4.3.13.RELEASE'
+  ext.testWithBootVersion = '1.5.9.RELEASE'
+  
   dependencies {
     // java-cloudant client
     compile group: 'com.cloudant', name: 'cloudant-client', version: '2.11.0'
     // spring
-    compile group: 'org.springframework', name: 'spring-core', version: springVersion
-    compile group: 'org.springframework', name: 'spring-context', version: springVersion
-    compile group: 'org.springframework', name: 'spring-beans', version: springVersion
-    // testing
+    compile group: 'org.springframework', name: 'spring-core', version: compileWithSpringVersion
```

Comment:
I didn't want to play with the deps declaration types, in case they were important for the final pom, all the changes I added were in keeping with the existing build implementation.